scripts/ae_train.py

```diff
@@ -0,0 +1,169 @@
+
+    # TODO: Find a better way to import API key (eventually remove comet.ml)
+    logger.info(comet_ml.get_comet_version())
+    experiment_recorder = experiment.record_experiment(api_key=os.environ["COMETML"])
+
+    ref_anat_img = nib.load(experiment_dict["ref_anat_fname"])
+    isocenter = compute_isocenter(ref_anat_img)
+    volume = compute_volume(ref_anat_img)
+
+    # Load tractograms
+    logger.info("Loading tractograms...")
+
+    data_manager = DataManager(experiment_dict, seed)
+
+    logger.info("Finished loading tractograms.")
+
+    (
+        train_loader,
+        valid_loader,
+        test_loader,
+        viz_loader,
+    ) = data_manager.setup_data_loader()
+
+    data_loaders = (train_loader, valid_loader, test_loader)
+
+    logger.info("Building model and trainer...")
+
+    trainer = Trainer(
+        experiment_dict,
+        experiment.experiment_dir,
+        device,
+        data_loaders,
+        (data_manager.point_dims, data_manager.num_points),
+        isocenter,
+        volume,
+        experiment_recorder,
+    )
+
+    logger.info("Finished building model and trainer.")
+
+    # Start training run
+    logger.info("Starting training...")
+    for epoch in range(1, experiment_dict["epochs"] + 1):
+        with experiment_recorder.train():
+            trainer.train(epoch)
+        with experiment_recorder.validate():
+            trainer.valid(epoch)
+
+        # Project the valid set
+        val_latent_vecs, val_classes = encode_data(
+            viz_loader,
+            device,
+            trainer.model,
+            limit_batch=experiment_dict["viz_num_batches"],
+        )
+        # Save the vectors (and the class of each streamline for easy plotting)
+        torch.save(
+            {"vecs": val_latent_vecs, "classes": val_classes},
+            f"{experiment.experiment_dir}/prediction_valid.pt",
+        )
+        # Plot the latent space
+        latent_plot_filename = plot_latent_space(
+            val_latent_vecs,
+            val_classes,
+            val_latent_vecs.shape[1],
+            f"{experiment.experiment_dir}/latent_valid_epoch{epoch}",
+            experiment_dict["rbx_classes"],
+        )
+        # Log the latent space plot to Comet
+        experiment_recorder.log_image(
+            latent_plot_filename, name="latent_umap", step=epoch
+        )
+
+    logger.info("Finished training.")
+    torch.cuda.empty_cache()
+    lowest_loss_epoch = trainer.load_checkpoint(trainer.best_checkpoint, device)
+
+    trainer.model.eval()
+    train_loss_data_fname, valid_loss_data_fname = trainer.save_loss_history()
+
+    loss_plot_fname = pjoin(trainer.run_dir, LoggerKeys.loss_plot_file_basename.value)
+    plot_loss_history(train_loss_data_fname, valid_loss_data_fname, loss_plot_fname)
+
+    logger.info(
+        "Loaded saved best model: loss {} at epoch: {}".format(
+            trainer.lowest_loss, lowest_loss_epoch
+        )
+    )
+
+    logger.info("Finished experiment.")
```

Comment:
In a second pass, we will be willing to refactor this entire block into methods that can be re-used, the `main()` virtually containing a few method calls and being as short as possible. Same applies to any other script.

Reply:
To do in another PR